cobra/core/configuration.py

```diff
@@ -94,15 +107,73 @@ def bounds(self):
     @bounds.setter
     def bounds(self, bounds):
         # TODO: We should consider allowing `None` for free bounds.
-        assert bounds[0] <= bounds[1]
-        self.lower_bound = bounds[0]
-        self.upper_bound = bounds[1]
+        if bounds[0] <= bounds[1]:
```

Comment:
Why do you suggest this change in behavior? I think that actually raising an exception is important rather than an error which can be missed and will lead to problems later.

Reply:
Oh yes! I totally forgot to mention that I was thinking that adding a new exception like `BoundsError` might be a better option than `AssertionError`. This will not only give a proper context for the error but also can be introduced to other parts of the code base which might require it.